app/views/shared/_header.html.haml

```diff
@@ -45,5 +61,81 @@
           = link_to t('navigation.sign_out'), destroy_user_session_path, method: 'delete', aria: { label: t('navigation.sign_out') }
       - else
         %li
-          %a{ href: new_user_session_path, aria: { label: t('users.login') } }
+          %a{ href: new_user_session_path, aria: { label: t('navigation.sign_in') } }
             = t('navigation.sign_in')
+%div#translation-bar-desktop
+  %span= t('navigation.translate_to')
+  %a.translate-link{ href: '#', data: { lang: 'en' }, aria: { label: 'Translate to English' } }
+    %img{ src: asset_path('language_flags/english_language_flag.png'), alt: 'English' }
+  %a.translate-link{ href: '#', data: { lang: 'ar' }, aria: { label: 'Translate to Arabic' } }
+    %img{ src: asset_path('language_flags/arabic_language_flag.png'), alt: 'Arabic' }
+  %a.translate-link{ href: '#', data: { lang: 'zh-CN' }, aria: { label: 'Translate to Chinese (Simplified)' } }
+    %img{ src: asset_path('language_flags/chinese_simplified_language_flag.png'), alt: 'Chinese (Simplified)' }
+  %a.translate-link{ href: '#', data: { lang: 'zh-TW' }, aria: { label: 'Translate to Chinese (Traditional)' } }
+    %img{ src: asset_path('language_flags/chinese_traditional_language_flag.png'), alt: 'Chinese (Traditional)' }
+  %a.translate-link{ href: '#', data: { lang: 'fr' }, aria: { label: 'Translate to French' } }
+    %img{ src: asset_path('language_flags/french_language_flag.png'), alt: 'French' }
+  %a.translate-link{ href: '#', data: { lang: 'ko' }, aria: { label: 'Translate to Korean' } }
+    %img{ src: asset_path('language_flags/korean_language_flag.png'), alt: 'Korean' }
+  %a.translate-link{ href: '#', data: { lang: 'es' }, aria: { label: 'Translate to Spanish' } }
+    %img{ src: asset_path('language_flags/spanish_language_flag.png'), alt: 'Spanish' }
+
+:javascript
+  (function() {
+    var currentLang = 'en';
+
+    function translatePage(lang) {
+      if (currentLang === lang) return;
+      
+      var elements = document.getElementsByTagName('*');
+      for (var i = 0; i < elements.length; i++) {
+        var element = elements[i];
+        if (element.id === 'map' || element.closest('#map') || element.classList.contains('map-related')) {
+          continue;
+        }
+        if (element.childNodes.length === 1 && element.childNodes[0].nodeType === Node.TEXT_NODE) {
+          translateElement(element, lang);
+        }
+      }
+      currentLang = lang;
+    }
+
+    function translateElement(element, lang) {
+      var text = element.textContent || element.innerText;
```

Comment:
While testing this locally I noticed that the placeholders are not translated: 
![image](https://github.com/user-attachments/assets/ea51cdbf-9320-4b8a-ab6a-f4495a07aea6)

![image](https://github.com/user-attachments/assets/0d22a017-971d-4640-9373-c853aedc7db8)

Maybe we need to add here a case to account for the placeholder before we sent the text to the API.